src/replayq_mem_ets_exclusive.erl

```diff
@@ -0,0 +1,115 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2025 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% @doc Exclusively owned ETS table for in-memory queue.
+-module(replayq_mem_ets_exclusive).
+
+-behaviour(replayq_mem).
+
+-export([
+    new/1,
+    peek_all/1,
+    peek/1,
+    is_empty/1,
+    in/2,
+    in_r/2,
+    in_batch/2,
+    out/1,
+    destroy/1
+]).
+
+-export_type([queue/0]).
+
+-opaque queue() :: #{ets_tab := ets:tab(), next_id := non_neg_integer()}.
+
+%% @doc Create a new queue.
+-spec new(_) -> queue().
+new(_) ->
+    Tab = ets:new(?MODULE, [ordered_set, public]),
+    #{ets_tab => Tab, next_id => 1}.
+
+%% @doc Peek all items from the queue.
+-spec peek_all(queue()) -> [term()].
+peek_all(#{ets_tab := Tab}) ->
+    lists:map(fun({_Id, Item}) -> Item end, ets:tab2list(Tab)).
+
+%% @doc Peek the front item of the queue.
+-spec peek(queue()) -> empty | {value, term()}.
+peek(#{ets_tab := Tab}) ->
+    case ets:first(Tab) of
+        '$end_of_table' ->
+            empty;
+        Id ->
+            Item = ets:lookup_element(Tab, Id, 2),
+            {value, Item}
+    end.
+
+%% @doc Return 'true' if the queue is empty.
+-spec is_empty(queue()) -> boolean().
+is_empty(#{ets_tab := Tab}) ->
+    ets:first(Tab) =:= '$end_of_table'.
+
+%% @doc Enqueue an item to the queue.
+-spec in(term(), queue()) -> queue().
+in(Item, #{ets_tab := Tab, next_id := NextId} = Q) ->
+    ets:insert(Tab, {NextId, Item}),
+    Q#{next_id => NextId + 1}.
+
+%% @doc Enqueue an item in reverse order to the queue.
+-spec in_r(term(), queue()) -> queue().
+in_r(Item, #{ets_tab := Tab} = Q) ->
+    case ets:first(Tab) of
+        '$end_of_table' ->
+            in(Item, Q);
+        Id ->
+            %% Id is the key of the first item in the queue
+            %% do not allow inverse order enqueue
+            Id =< 1 andalso erlang:error(badarg),
```

Comment:
actually, this module doesn't even need this: it doesn't matter if it's negative here (differently from the `shared` flavor)